pydantic/config.py

```diff
@@ -709,11 +709,35 @@ class Model(BaseModel):
 
     defer_build: bool
     """
-    Whether to defer model validator and serializer construction until the first model validation.
+    Whether to defer model validator and serializer construction until the first model validation. Defaults to False.
 
     This can be useful to avoid the overhead of building models which are only
     used nested within other models, or when you want to manually define type namespace via
-    [`Model.model_rebuild(_types_namespace=...)`][pydantic.BaseModel.model_rebuild]. Defaults to False.
+    [`Model.model_rebuild(_types_namespace=...)`][pydantic.BaseModel.model_rebuild].
+
+    See also [`_defer_build_mode`][pydantic.config.ConfigDict._defer_build_mode].
+
+    !!! note
+        `defer_build` does not work by default with FastAPI Pydantic models. By default, the validator and serializer
+        for said models is constructed immediately for FastAPI routes. You also need to define
+        [`_defer_build_mode=('model', 'type_adapter')`][pydantic.config.ConfigDict._defer_build_mode] with FastAPI
+        models in order for `defer_build=True` to take effect. This additional (experimental) parameter is required for
+        the deferred building due to FastAPI relying on `TypeAdapter`s.
+    """
+
+    _defer_build_mode: tuple[Literal['model', 'type_adapter'], ...]
```

Comment:
Noticed that `pydantic.dataclasses.dataclass` doesn't either work with `defer_build`. So in some future PR the `_defer_build_mode` could include `dataclass`